github_app.go

```diff
@@ -91,32 +94,31 @@ func (s *Settings) CreateInstallationTokenWithDefaultOwnerRepo(ctx context.Conte
 	if s.AuthConfig.Github == nil || s.AuthConfig.Github.DefaultOwner == "" || s.AuthConfig.Github.DefaultRepo == "" {
 		return "", errors.Errorf("missing GitHub app configuration needed to create installation tokens")
 	}
-	return s.CreateInstallationToken(ctx, s.AuthConfig.Github.DefaultOwner, s.AuthConfig.Github.DefaultRepo, opts)
+	return CreateInstallationToken(ctx, s.CreateGitHubAppAuth(), s.AuthConfig.Github.DefaultOwner, s.AuthConfig.Github.DefaultRepo, opts)
 }
 
 // CreateInstallationToken uses the owner/repo information to request an github app installation id
 // and uses that id to create an installation token.
-func (s *Settings) CreateInstallationToken(ctx context.Context, owner, repo string, opts *github.InstallationTokenOptions) (string, error) {
-	authFields := getGithubAppAuth(s)
-	if authFields == nil {
+func CreateInstallationToken(ctx context.Context, githubAppAuth *GithubAppAuth, owner, repo string, opts *github.InstallationTokenOptions) (string, error) {
```

Comment:
now that CreateInstallationToken and HasGitHubApp are now closely tied to the GithubAppAuth structure, we should consider making the functions methods of GithubAppAuth.